ProcessLib/UncoupledProcessesTimeLoop.cpp

```diff
@@ -838,6 +846,12 @@ bool UncoupledProcessesTimeLoop::solveCoupledEquationSystemsByStaggeredScheme(
             ++process_id;
             continue;
         }
+        CoupledSolutionsForStaggeredScheme coupled_solutions(
+            _solutions_of_coupled_processes, dt, process_id);
+
+        process_data->process.setCoupledSolutionsForStaggeredScheme(
+            &coupled_solutions);
```

Comment:
`coupled_solutions` is a locally created object. I wonder if giving the address of a local created object to the method `setCoupledSolutionsForStaggeredScheme()` is a good idea?

Reply:
Exactly this is the bug, unfortunately. The reference can be easily stolen from the `process_data`.
The same code is in the assemble loop and was missing in the post timestep part.